src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -108,11 +109,14 @@ public override async Task ExecuteCommandAsync()
                 var v2RestoreResult = await PerformNuGetV2RestoreAsync(restoreInputs);
                 restoreSummaries.Add(v2RestoreResult);
 
-                // log warnings
-                v2RestoreResult
-                    .Errors
-                    .Where(l => l.Level == LogLevel.Warning)
-                    .ForEach(l => Console.LogWarning(l.FormatWithCode()));
+                // log warnings for a failure since the package extractor throws an exception without logging warnings
+                if (!v2RestoreResult.Success)
```

Comment:
I don't follow the comment and the code change... you're skipping logging warning in success case but the comment says completely different thing